src/services/CertificateGenerationService.ts

```diff
@@ -177,6 +171,39 @@ class CertificateGenerationService {
             });
     }
 
+    /**
+     * Method to check if welsh feature flags are enabled and check if test station has a Welsh postcode
+     * @param testResult
+     * @param flags
+     * @returns boolean for if test station is Welsh
+     */
+    public async isTestStationWelsh(testResult: any, flags: any): Promise<boolean> {
+        try {
+            // Execute bilingual certificate generation and logic if FF enabled
+            if (!flags.welshTranslation.enabled) {
+                throw new Error(`Unable to translate test result: global Welsh translation flag disabled.`);
+            }
+            const shouldTranslate: boolean =
+                (testResult.testTypes.testResult === "prs" && flags.welshTranslation.translatePrsTestResult) ||
+                (testResult.testTypes.testResult === "pass" && flags.welshTranslation.translatePassTestResult) ||
+                (testResult.testTypes.testResult === "fail" && flags.welshTranslation.translateFailTestResult);
+
+            if (shouldTranslate) {
+                // Find out if Welsh certificate is required
+                const testStations = await this.getTestStations();
+                const testStationPostcode = this.getThisTestStation(testStations, testResult.testStationPNumber);
+                return testStationPostcode ? await this.lookupPostcode(testStationPostcode) : false;
+            } else {
+                throw new Error(`flag disabled for test result: ${testResult.testTypes.testResult}`);
```

Comment:
why is the throw needed? You're throwing it, and it's being caught locally

Reply:
Used throwing to break the flow in places and jump to the catch to display the error & return false.

Reply:
Agree with Dan, if there is a way of applying the same logic with console.log / console.warn rather than throwing and catching an Error (to do a log) I think it would look cleaner.